pkgs/tools/text/gnupatch/default.nix

```diff
@@ -11,6 +11,12 @@ stdenv.mkDerivation rec {
     sha256 = "1zfqy4rdcy279vwn2z1kbv19dcfw25d2aqy9nzvdkq5bjzd0nqdc";
   };
 
+  patches = fetchurl {
+    name = "CVE-2018-6951.patch";
+    url = "https://git.savannah.gnu.org/cgit/patch.git/patch/?id=f290f48a621867084884bfff87f8093c15195e6a";
+    sha256 = "0289g8ihyiqdxw4p3zkal7ivbpy4dp5s31x2vy9ajw67q0694ywa";
+  };
```

Comment:
It is better to add the patch to nixpkgs instead + add the source url as a comment. There is a version number of cgit in the footer that will eventually break the hash.

Reply:
Done.